lib/pypechain/pypechain/utilities/abi.py

```diff
@@ -0,0 +1,86 @@
+"""Utilities for solidity contract ABIs."""
+from __future__ import annotations
+
+import json
+from typing import List, NamedTuple, Optional, TypeGuard
+
+from web3.types import ABIEvent, ABIFunction
+
+
+class Input(NamedTuple):
+    """An input of a function or event."""
+
+    internalType: str
+    name: str
+    type: str
+    indexed: Optional[bool] = None
```

Comment:
I believe the `Optional[]` syntax is being deprecated in favor of `indexed: bool | None = None`

Reply:
oh yeah thanks.